nbconvert/filters/markdown_mistune.py

```diff
@@ -21,6 +21,12 @@
 from nbconvert.filters.strings import add_anchor
 
 
+class MathBlockGrammar(mistune.BlockGrammar):
+    block_math = re.compile(r"^\$\$(.*?)\$\$|^\\\\\[(.*?)\\\\\]", re.DOTALL)
+    latex_environment = re.compile(r"^\\begin\{([a-z]*\*?)\}(.*?)\\end\{\1\}",
```

Comment:
This should not exist as a separate regular expression. 
It should be included in the previous one(suggested to be called `multiline_math`) as an alternation.

Reply:
Additionally, once it is combined with the previous one, it should include its delimiters only grab a single group.